### PR TITLE
Fix: Duplicated SSO login 

### LINKED ISSUE #1460: Deep links. Return to app second time will lead to error.
## Description

When user click on "Open Nova Spektr.app" second time it will lead to log out and show the error.

## Steps to Reproduce

1. Go to matrix log in flow
2. Login via GitHub
3. Open GitHub link again

## Expected Behavior
Show message, something like "You are already logged in"

## Actual Behavior
Log out from matrix and show an error, can't log in again via old browser window

## Environment

- Operating system: macOS
- Version: Stage build -  Nova-Spektr-Stage-1.4.4-3-arm64.dmg

## Screenshots
Screenshot 2024-02-15 at 12 52 24

logs:
[nova-spektr.log](https://github.com/novasamatech/nova-spektr/files/14294915/nova-spektr.log)

### PR DESCRIPTION
**Pull request by tuul-wq**
- Try to login with token if token exists and cache failed
- Use $loginStatus in MatrixLogin form
- Fixed stage dev for electron
- Tests

closes #1460

### CODE REVIEW

File: src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts [hunk 1]
```diff
@@ -25,68 +26,90 @@ const loginFromCacheFx = createEffect((matrix: ISecureMessenger): Promise<void>
...
-    withToken: (value) => Boolean(value),
-  },
-  cases: {
-    withToken: loggedInWithToken,
-    __: loggedInFromCache,
+  clock: loggedInFromCache,
+  source: {
+    matrix: matrixModel.$matrix,
+    loginStatus: matrixModel.$loginStatus,
+    autoLoginStatus: $autoLoginStatus,
```
Comment (Asmadek) [inline src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts:33]: Why do you need `autoLoginStatus`? as i see you don't use in filter or fn

File: src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts [hunk 2]
```diff
@@ -25,68 +26,90 @@ const loginFromCacheFx = createEffect((matrix: ISecureMessenger): Promise<void>
...
+    loginStatus: matrixModel.$loginStatus,
+  }),
 });
 
 sample({
-  clock: [loggedInWithToken, loggedInFromCache],
-  fn: () => LoginStatus.AUTO_LOGIN,
-  target: matrixModel.events.loginStatusChanged,
+  clock: loggedInWithToken,
+  filter: loginFromCacheFx.pending,
```
Comment (Asmadek) [inline src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts:48]: new use case for filter, I will remember it 👍

File: src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts [hunk 3]
```diff
@@ -25,68 +26,90 @@ const loginFromCacheFx = createEffect((matrix: ISecureMessenger): Promise<void>
...
 
 sample({
-  clock: loggedInFromCache,
-  source: matrixModel.$matrix,
-  target: loginFromCacheFx,
+  clock: proceedWithToken,
+  source: {
+    matrix: matrixModel.$matrix,
+    loginStatus: matrixModel.$loginStatus,
+    autoLoginStatus: $autoLoginStatus,
```
Comment (Asmadek) [inline src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts:63]: I am not sure it needed here too

**Review by Asmadek — APPROVED** [review]

File: src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts (hunk 2 above)
Comment (tuul-wq) [reply src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts:48]: Yeah, someone told me about this in effector chat

File: src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts (hunk 1 above)
Comment (tuul-wq) [reply src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts:33]: We use it in `ui/MatrixAutoLogin` to show animation status

**Pull request merged**